src/Microsoft.AspNet.Hosting/WebApplicationConfiguration.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.AspNet.Hosting
+{
+    public class WebApplicationConfiguration
+    {
+        public static readonly string ApplicationKey = "app";
+        public static readonly string DetailedErrorsKey = "detailedErrors";
+        public static readonly string EnvironmentKey = "environment";
+        public static readonly string ServerKey = "server";
+        public static readonly string WebRootKey = "webroot";
+        public static readonly string CaptureStartupErrorsKey = "captureStartupErrors";
+
+        public static readonly string HostingJsonFile = "hosting.json";
+        public static readonly string EnvironmentVariablesPrefix = "ASPNET_";
+
+        public static IConfiguration GetDefault()
+        {
+            return GetDefault(args: null);
```

Comment:
We shouldn't assume any of the configuration sources are "hot" by default when self-hosting through the builder. See other remark for adding builder.UseConfiguration where you opt-in and control which json file and environment variables (if any) apply to your program.

```cs
        public WebApplicationBuilder UseConfiguration(
            string hostingJsonFile = null,
            string environmentVariablesPrefix = null,
            string[] commandLineArgs = null)
```